src/lib.rs

```diff
@@ -132,21 +151,47 @@ macro_rules! define_vector {
     };
 }
 
-define_vector!(Vec2, Vector2, 8, f32, 2 <- "Vector of 2 f32s. Alignment 8, size 16.");
-define_vector!(Vec3, Vector3, 16, f32, 3 <- "Vector of 3 f32s. Alignment 16, size 24.");
-define_vector!(Vec4, Vector4, 16, f32, 4 <- "Vector of 4 f32s. Alignment 16, size 32.");
-define_vector!(DVec2, Vector2, 16, f64, 2 <- "Vector of 2 f64s. Alignment 16, size 32.");
-define_vector!(DVec3, Vector3, 32, f64, 3 <- "Vector of 3 f64s. Alignment 32, size 48.");
-define_vector!(DVec4, Vector4, 32, f64, 4 <- "Vector of 4 f64s. Alignment 32, size 64.");
-define_vector!(UVec2, Vector2, 8, u32, 2 <- "Vector of 2 u32s. Alignment 8, size 16.");
-define_vector!(UVec3, Vector3, 16, u32, 3 <- "Vector of 3 u32s. Alignment 16, size 24.");
-define_vector!(UVec4, Vector4, 16, u32, 4 <- "Vector of 4 u32s. Alignment 16, size 32.");
-define_vector!(IVec2, Vector2, 8, i32, 2 <- "Vector of 2 i32s. Alignment 8, size 16.");
-define_vector!(IVec3, Vector3, 16, i32, 3 <- "Vector of 3 i32s. Alignment 16, size 24.");
-define_vector!(IVec4, Vector4, 16, i32, 4 <- "Vector of 4 i32s. Alignment 16, size 32.");
-
-macro_rules! define_matrix {
-    ($name:ident, $mint_type:ident, $align:literal, $inner_ty:ty, $ty:ty, $count_x:literal, $count_y:literal, $padding:literal -> $($idx:literal),* <- $doc:literal) => {
+define_vectors! {
+    (Vec2, Vector2, f32 * 2, align: 8, size: 16),
+    (Vec3, Vector3, f32 * 3, align: 16, size: 24),
+    (Vec4, Vector4, f32 * 4, align: 16, size: 32),
+    (DVec2, Vector2, f64 * 2, align: 16, size: 32),
+    (DVec3, Vector3, f64 * 3, align: 32, size: 48),
+    (DVec4, Vector4, f64 * 4, align: 32, size: 64),
+    (UVec2, Vector2, u32 * 2, align: 8, size: 16),
+    (UVec3, Vector3, u32 * 3, align: 16, size: 24),
+    (UVec4, Vector4, u32 * 4, align: 16, size: 32),
+    (IVec2, Vector2, i32 * 2, align: 8, size: 16),
+    (IVec3, Vector3, i32 * 3, align: 16, size: 24),
+    (IVec4, Vector4, i32 * 4, align: 16, size: 32),
+}
+
+macro_rules! define_matrices {
+    ( $(( $name:ident, $mint_name:ident, $prim_ty:ty, $row_ty:ty, $rows:literal * $cols:literal, align: $align:literal, size: $size:literal, pad: $pad:literal, [$($idx:literal),*] ),)* ) => {
+        $(
+            define_matrices!(@impl
+                $name,
+                mint::$mint_name<$prim_ty>,
+                $align,
+                $prim_ty,
+                $row_ty,
+                $rows,
+                $cols,
+                $pad,
+                [$( $idx ),*],
+                concat!(
+                    "Matrix of `", stringify!($prim_ty), "` with ", stringify!($rows), " rows and ", stringify!($cols), " columns. ",
+                    "Has size ", stringify!($size), ", alignment ", stringify!($align), ", and ", stringify!($pad), " bytes of extra padding."
```

Comment:
Let's remove the "X bytes of extra padding" as it doesn't really help users, and just clutters up the docs.